src/StatusCodeException/InvalidStatusCodeException.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Exception thrown when the Status Code is invalid.
+ *
+ * PHP version 5.3
+ *
+ * @category Exception
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot\StatusCodeException;
+
+use InvalidArgumentException;
+
+/**
+ * Exception thrown when the Status Code is invalid.
+ *
+ * PHP version 5.3
+ *
+ * @category Exception
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class InvalidStatusCodeException extends InvalidArgumentException
+{
+    /**
+     * Named constructor for non-numeric status codes.
+     *
+     * @param mixed $code the non-numeric code
+     * @return InvalidStatusCodeException
+     */
+    public static function notNumeric($code)
+    {
+        return new self(sprintf(
```

Comment:
@shrikeh 

Nice, glad you liked it!

Actually picked this up from Doctrine2, and I like it.